bip-0370.mediawiki

```diff
@@ -213,11 +213,15 @@ The nLockTime field of a transaction is determined by inspecting the PSBT_GLOBAL
 If none of the inputs have a PSBT_IN_REQUIRED_TIME_LOCKTIME and PSBT_IN_REQUIRED_HEIGHT_LOCKTIME, then PSBT_GLOBAL_FALLBACK_LOCKTIME must be used.
 If PSBT_GLOBAL_FALLBACK_LOCKTIME is not provided, then it is assumed to be 0.
 
-If one or more inuts have a PSBT_IN_REQUIRED_TIME_LOCKTIME or PSBT_IN_REQUIRED_HEIGHT_LOCKTIME, then the field chosen is the one which is supported by all of the inputs.
+If one or more inputs have a PSBT_IN_REQUIRED_TIME_LOCKTIME or PSBT_IN_REQUIRED_HEIGHT_LOCKTIME, then the field chosen is the one which is supported by all of the inputs.
 This can be determined by looking at all of the inputs which specify a locktime in either of those fields, and choosing the field which is present in all of those inputs.
 Inputs not specifying a lock time field can take both types of lock times, as can those that specify both.
 The lock time chosen is then the maximum value of the chosen type of lock time.
 
+If a PSBT has both types of locktimes possible, then locktime determined by looking at the PSBT_IN_REQUIRED_HEIGHT_LOCKTIME fields of the inputs must be chosen.<ref>'''Why choose the height based locktime?'''
```

Comment:
Suggest: "If a PSBT has both types of locktime possible because one or more inputs specify both PSBT_IN_REQUIRED_TIME_LOCKTIME and PSBT_IN_REQUIRED_HEIGHT_LOCKTIME, ..."

The case where both types are implicitly possible by virtue of not being provided should fall back to PSBT_GLOBAL_FALLBACK_LOCKTIME or 0.

Reply:
Done